fast-af/fast-checkout-woocommerce
# PINT-493 - Add product options to headless PDP buttons

# Description

Update the shortcode, widget, and Gutenberg block for the PDP button to include the `product_options` parameter. Also, update the Gutenberg PDP button block to enable searching for and selecting products and variations instead of having to enter product and variation ID's.

See https://fastaf.atlassian.net/browse/PINT-493

# Testing instructions

1. Login to the [Fast WC dev admin](https://fasttestdev.wpcomstaging.com/wp-admin).
2. Go to Posts or Pages.
3. Add a new post or page.
4. In the Gutenberg editor, click the `+` button to add a block.
5. Search for the Fast Checkout Product Button block.
6. In the right sidebar under Product Details, search for a product to add. (Note, the "Hoodie" and "Beanie" products have variations that you can test.)
7. Select the product you want to use.
8. If the product has variations, select the variation you want to use.
9. If the variation has selectable Product Options, select the product options you want to use.
10. Publish the post or page.
11. View the post or page on the front end and verify that the button works as expected.

# Checks

@ilkerulutas @brikr this is ready for code review.
